docs/rfc/derived-aspects/derived-aspects.md

```diff
@@ -0,0 +1,167 @@
+In this feature we will discuss how we can enable search and discovery on the derived metadata. 
+
+## Basic example
+
+Metadata `Health` of a dataset is a typical example of this feature. The details of metadata `Health` will be covered in a separate RFC (the review of `Health` metadata is out of scope of this RFC)
+
+To put it simple,  
+- Health metadata is derived using metadata from multiple aspects.
+- A need for promoting healthy datasets in search of the datasets.
+- There are organizational needs to rollup unhealthy (or healthy) datasets to a ldap/manager/team. This has the needs to query graph in conjunction with health metadata.
+    - Number of unhealthy datasets owned by me.
+    - Number of unhealthy datasets owned by my organization.
+
+## Motivation
+In the context of derived metadata, we need the ability to 
+1. Compute from different aspects
+2. Maintain versions of the derived metadata 
+3. Notify in the event of change in derived metadata
+4. Search of entities powered with derived metadata
+5. Graph query needs of entities powered with derived metadata.
+6. Dump derived metadata along with regular aspects for offline consumption / analysis.
+
+## Detailed design
+
+Derived metadata can be modelled as an aspect of an entity. This is similar to any other aspect but with subtle differences.
+
+- Value of this aspect gets computed only when `GET` API is invoked. 
+- `GET` API computes, persists and retrieves the metadata from the KV store.
+
+### Entity Aspect Model
+
+Below shows a typical entity - aspect model for derived metadata. 
+
+![Derived Aspect Entity Aspect Model](Entity-Aspect-Model.png)
+ 
+In the above, `DerivedAspect` is an aspect of entity1. The value of this is computed using metadata from 
+- `aspect2` of `entity1`
+- `aspect3` of `entity2`
+- `other metadata` from auxiliary files / other services.
+
+### Value Computers
+Value computer classes computes the value of the derived aspects at that point of time by GMA DAOs.
+- GMA DAO will have the provision to register value computers. Similar to how one registers the post update hooks in creation of DAO instance.
+- GMA DAO on get of the aspect, will invoke the value computer, computes value, persists if necessary and returns the value.
+
+### Annotating derived aspects
+Derived aspects will be annotated in GMA using annotation `@gma.derivedaspect`. Refer below as an example
+
+```pdl
+@gma.derivedaspect
+record FooDerivedAspect {
+  barField : string
+}
+```
+These annotations will by used `GMA DAO` to ensure there is only process of computing / persisting the derived aspetcs.
+- derived aspects requires `ValueComputer` to be registered with DAO and will be validated
+- get method of derived aspect always recomputes the value using `ValueComputer`
+- add method of derived aspect will be blocked on the client provided value. It is always to be computed using `ValueComputer`.
```

Comment:
none of these steps seem to actually look at the annotation? its purpose is unclear to me.